libafl/src/observers/cmp.rs

```diff
@@ -35,6 +35,39 @@ where
     fn add_from(&mut self, usable_count: usize, cmp_map: &mut CM, cmp_observer_data: Self::Data);
 }
 
+/// A bytes string for cmplog with up to 32 elements.
+#[derive(Debug, Copy, Clone, Serialize, Deserialize, Eq, PartialEq)]
+pub struct CmplogBytes {
+    buf: [u8; 32],
+    len: u8,
+}
+
+impl CmplogBytes {
+    /// Creates a new [`CmplogBytes`] object from the provided buf and length.
+    /// Lengths above 32 are illegal but will be ignored.
+    #[must_use]
+    pub fn from_buf_and_len(buf: [u8; 32], len: u8) -> Self {
+        debug_assert!(len <= 32, "Len too big: {len}, max: 32");
+        CmplogBytes { buf, len: len % 32 }
```

Comment:
`% 32` breaks here

Reply:
`(len - 1) % 32`?

Reply:
Wait no :D

Reply:
`((len - 1 & 31) + 1) & len` lol... but at this point what's the point.